nixos/modules/services/web-apps/invidious.nix

```
@@ -7,29 +7,60 @@ let
 
   settingsFile = settingsFormat.generate "invidious-settings" cfg.settings;
 
+  finalHmacKeyFile =
+    if cfg.hmacKeyFile != null
+    then cfg.hmacKeyFile
+    else "/var/lib/invidious/hmac_key";
```

```suggestion
  generatedHmacKeyFile = "/var/lib/invidious/hmac_key";
```

This value is only ever used when `hmacKeyFile` is null.